### ISSUE TITLE
Check failed in DocDB compaction filter on distributed transaction metadata

### ISSUE THREAD

**Issue by robertpang** [YugaByte/yugabyte-db#197]
When DocDB compaction happens and encounters a key for distributed transaction metadata, check fails and the compaction aborts:

```F0420 13:33:28.525691 173228032 docdb_compaction_filter.cc:70] Check failed: key_decode_status.ok() Error decoding a key during compaction: Corruption (yb/docdb/doc_key.cc:47): Error when decoding range components of a document key: Unexpected end of key when decoding document key
    Key (raw): "\x0ax2y\xa4\xc6\xe4\xc9O\xad\x8e\x8d\x87\xde\x9a\xbbj\xfc"
    Key (best-effort decoded): "\x0ax2y\xa4\xc6\xe4\xc9O\xad\x8e\x8d\x87\xde\x9a\xbbj\xfc"
Fatal failure details written to /tmp/yugabyte-local-cluster/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2018-04-20T13_33_28.pid74369.txt
F20180420 13:33:28 ../../../../../src/yb/docdb/docdb_compaction_filter.cc:70] Check failed: key_decode_status.ok() Error decoding a key during compaction: Corruption (yb/docdb/doc_key.cc:47): Error when decoding range components of a document key: Unexpected end of key when decoding document key
    Key (raw): "\x0ax2y\xa4\xc6\xe4\xc9O\xad\x8e\x8d\x87\xde\x9a\xbbj\xfc"
    Key (best-effort decoded): "\x0ax2y\xa4\xc6\xe4\xc9O\xad\x8e\x8d\x87\xde\x9a\xbbj\xfc"
    @        0x1094e5c8b  google::LogDestination::LogToSinks()
    @        0x1094e4daf  google::LogMessage::SendToLog()
    @        0x1094e5775  google::LogMessage::Flush()
    @        0x1094e973f  google::LogMessageFatal::~LogMessageFatal()
    @        0x1094e63b9  google::LogMessageFatal::~LogMessageFatal()
    @        0x1081b96f2  yb::docdb::DocDBCompactionFilter::Filter()
    @        0x1083a8c8f  rocksdb::CompactionIterator::NextFromInput()
    @        0x1083a96a5  rocksdb::CompactionIterator::Next()
    @        0x1083aefa8  rocksdb::CompactionJob::ProcessKeyValueCompaction()
    @        0x1083ad7c0  rocksdb::CompactionJob::Run()
    @        0x1083e9057  rocksdb::DBImpl::BackgroundCompaction()
    @        0x1083e7950  rocksdb::DBImpl::BackgroundCallCompaction()
    @        0x108583cf9  rocksdb::ThreadPool::BGThread()
    @        0x1085840a3  rocksdb::BGThreadWrapper()
    @     0x7fff7c7a1661  _pthread_body
    @     0x7fff7c7a150d  _pthread_start
    @     0x7fff7c7a0bf9  thread_start

*** Check failure stack trace: ***
    @        0x1094e600a  google::LogMessage::Fail()
    @        0x1094e5058  google::LogMessage::SendToLog()
    @        0x1094e5775  google::LogMessage::Flush()
    @        0x1094e973f  google::LogMessageFatal::~LogMessageFatal()
    @        0x1094e63b9  google::LogMessageFatal::~LogMessageFatal()
    @        0x1081b96f2  yb::docdb::DocDBCompactionFilter::Filter()
    @        0x1083a8c8f  rocksdb::CompactionIterator::NextFromInput()
    @        0x1083a96a5  rocksdb::CompactionIterator::Next()
    @        0x1083aefa8  rocksdb::CompactionJob::ProcessKeyValueCompaction()
    @        0x1083ad7c0  rocksdb::CompactionJob::Run()
    @        0x1083e9057  rocksdb::DBImpl::BackgroundCompaction()
    @        0x1083e7950  rocksdb::DBImpl::BackgroundCallCompaction()
    @        0x108583cf9  rocksdb::ThreadPool::BGThread()
    @        0x1085840a3  rocksdb::BGThreadWrapper()
    @     0x7fff7c7a1661  _pthread_body
    @     0x7fff7c7a150d  _pthread_start
    @     0x7fff7c7a0bf9  thread_start```

**Comment by robertpang** [comment]
This happens when running the CassandraTransactionalKeyValue write-only workload. The workload was first run once with 128 writers and terminated after about 200k keys were inserted. Then the workload was rerun and the error happened after about 79k keys were re-inserted.

**Comment by robertpang** [comment]
Fixed in commit [e86a336](https://github.com/YugaByte/yugabyte-db/commit/e86a3369645bca93b4600a2d4d32f3f47cf21f86).
